utils/azure/storage.py

```diff
@@ -25,3 +25,33 @@ def instantiate_blob_client(
     blob_service_client = BlobServiceClient(account_url, credential=sp_credential)
 
     return blob_service_client
+
+
+def get_unique_jobs_from_blobs(blob_list: list = []) -> list:
```

Comment:
```suggestion
def get_unique_jobs_from_blobs(blob_list: list | None = None) -> list:
```
I would suggest not using an empty list as the default argument. [This is](https://stackoverflow.com/questions/366422/how-can-i-avoid-issues-caused-by-pythons-early-bound-default-parameters-e-g-m) one of the weirder footguns in python.

Reply:
Oh interesting! I was unaware of this; thanks for the heads up 👍